dao/mongo.go

```diff
@@ -259,52 +261,148 @@ func (m *MongoService) DeletePractitioner(practitionerID string, transactionID s
 		return http.StatusInternalServerError, fmt.Errorf("there was a problem handling your request for transaction id %s", transactionID)
 	}
 
-	// Choose specific practitioner to delete
-	pullQuery := bson.M{"data.practitioners": bson.M{"id": practitionerID}}
+	err = storedInsolvency.Decode(&insolvencyResource)
+	if err != nil {
+		log.Error(err)
+		return http.StatusInternalServerError, err
+	}
 
-	update, err := collection.UpdateOne(context.Background(), filter, bson.M{"$pull": pullQuery})
+	//get insolvency practitioners
+	mappedInsolvencyPractitioners, _, err := utils.ConvertStringToMapObjectAndStringList(insolvencyResource.Data.Practitioners)
 	if err != nil {
 		log.Error(err)
-		return http.StatusInternalServerError, fmt.Errorf("there was a problem handling your request for transaction id %s - could not delete practitioner with id %s", transactionID, practitionerID)
+		return http.StatusInternalServerError, fmt.Errorf("there was a problem handling your request for transaction id %s not able to map insolvency practitioners", transactionID)
 	}
 
-	// Return error if Mongo could not update the document
-	if update.ModifiedCount == 0 {
-		err = fmt.Errorf("there was a problem handling your request for transaction id %s - practitioner with id %s not found", transactionID, practitionerID)
+	// delete slice out the practitioner to delete from the map before updating insolvency
+	delete(mappedInsolvencyPractitioners, practitionerID)
```

Comment:
Would this not cause panic if the practitioner_id does not exist for the transaction?  We also don't seem to have a validation of the practitioner_id against the transaction_id.

Reply:
delete() should be fine if no practitioner id in the map ("If m is nil or there is no such element, delete is a no-op." - https://pkg.go.dev/builtin#delete).

Agree that we need to consider txn id validation (here & for other operations), perhaps as a follow-up?